fili-generic-example/src/main/java/com/yahoo/wiki/webservice/data/config/auto/DruidNavigator.java

```diff
@@ -0,0 +1,214 @@
+    private final DruidWebService druidWebService;
+    private final List<TableConfig> tableConfigurations;
+    private final CountDownLatch countDownLatch;
+
+    /**
+     * Constructs a DruidNavigator to load datasources from druid.
+     *
+     * @param druidWebService The DruidWebService to be used when talking to druid.
+     */
+    public DruidNavigator(DruidWebService druidWebService) {
+        this.druidWebService = druidWebService;
+        tableConfigurations = new ArrayList<>();
+        countDownLatch = new CountDownLatch(1);
+    }
+
+    /**
+     * Queries druid for all datasources and loads each of their configurations.
+     *
+     * @return a list of {@link DataSourceConfiguration} containing all known druid datasources configs.
+     */
+    @Override
+    public List<? extends DataSourceConfiguration> get() {
+        if (tableConfigurations.isEmpty()) {
+            loadAllDatasources();
+            LOG.info("Loading all datasources");
+            try {
+                //TODO Druid should ideally return a future, but this accomplishes the same result.
+                countDownLatch.await(5, TimeUnit.SECONDS);
+            } catch (InterruptedException e) {
+                LOG.error("Interrupted while waiting for a response from druid", e);
+                throw new RuntimeException("Unable to automatically configure correctly", e);
+            }
+        }
+
+        return tableConfigurations;
+    }
+
+    /**
+     * Queries druid for all datasources and loads all of the discovered tables.
+     *
+     * The expected response is:
+     * ["wikiticker"]
+     */
+    private void loadAllDatasources() {
+        queryDruid(rootNode -> {
+            if (rootNode.isArray()) {
+                rootNode.forEach(jsonNode -> {
+                    TableConfig tableConfig = new TableConfig(jsonNode.asText());
+                    loadTable(tableConfig);
+                    tableConfigurations.add(tableConfig);
+                });
+            }
+        }, COORDINATOR_TABLES_PATH);
+    }
+
+    /**
+     * Load a specific table with all reported metrics, dimensions, and timegrains.
+     * The expected response is:
+     * {
+     *      ...
+     *      "segments": [
+     *          {
+     *              "dataSource": "wikiticker",
+     *              "interval": "2015-09-12T00:00:00.000Z/2015-09-13T00:00:00.000Z",
+     *              ...,
+     *              "dimensions": "channel,cityName,comment,...",
+     *              "metrics": "count,added,deleted,delta,user_unique",
+     *              ...
+     *          }
+     *      ]
+     * }
+     *
+     * @param table The TableConfig to be loaded with queries against druid.
+     */
+    private void loadTable(TableConfig table) {
+        String url = COORDINATOR_TABLES_PATH + table.getName() + "/?full";
+        queryDruid(rootNode -> {
+            JsonNode segments = rootNode.get("segments").get(0);
+            loadMetrics(table, segments);
```

Comment:
Make a note in class comments that the schema of the first segment is for configuration.